### PR TITLE
chore: remove the redundant sns--test_feature target

### PR DESCRIPTION
**Pull request by rumenov**
`sns--test_feature` is supposed to enable a feature called "`test`" however, when grep-ing the code under `/rs/sns/cli` I don't find 
`[cfg(feature = "test")]  `

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

**Comment by max-dfinity** [comment]
The test features are rather in the dependencies included, not in the sns cli itself.  This was originally developed to make it possible to run test commands against test canisters when there were API changes.  

I'm not sure if it's still needed (and my preference is we get rid of "test_feature" dependency propagation, and instead find a more localized way to set flags).

**Review by anchpop — APPROVED** [review]

File: rs/sns/cli/BUILD.bazel [hunk 1]
```diff
@@ -72,23 +72,6 @@ rust_library(
...
-            "main.rs",
-        ],
-    ),
-    aliases = ALIASES,
-    crate_features = ["test"],
-    crate_name = "ic_sns_cli",
-    proc_macro_deps = MACRO_DEPENDENCIES,
-    version = "1.0.0",
-    deps = DEPENDENCIES_WITH_TEST_FEATURES,
-)
```
Comment (anchpop) [inline rs/sns/cli/BUILD.bazel:90]: Do we still have a use case where we need a test version of the SNS ClI?

Comment (max-dfinity) [reply rs/sns/cli/BUILD.bazel:90]: @aterga will be able to respond to this when he's back.  I don't recall the use case.  I would like to remove all of the test_feature stuff, as it's becoming quite a hassle when breaking up canister dependencies (in another PR)

Comment (rumenov) [reply rs/sns/cli/BUILD.bazel:90]: Just looking at the code and seeing there 
there are no `[cfg(feature = "test")]  nor [cfg(not(feature = "test"))]`  macros under the cli directory

the PR seems safe

Comment (max-dfinity) [reply rs/sns/cli/BUILD.bazel:90]: The PR is safe in the context of the IC monorepo.

The test version of the binary that can be used externally would go away, and that could cause problems in other testing scenarios.  DEPENDENCIES_WITH_TEST_FEATURES means that the test version of the binary is compiled against libraries with test features.  There were some behavioral differences at one point that were relevant for testing.  I want to make sure we don't break NNS Dapp testing workflows that depended on that.

Comment (rumenov) [reply rs/sns/cli/BUILD.bazel:90]: I completely missed DEPENDENCIES_WITH_TEST_FEATURES

**Pull request closed without merging**